Ponca/src/Fitting/linePrimitive.h

```diff
@@ -0,0 +1,139 @@
+/*
+ This Source Code Form is subject to the terms of the Mozilla Public
+ License, v. 2.0. If a copy of the MPL was not distributed with this
+ file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+
+
+#pragma onces
+
+#include "./defines.h"
+#include "./primitive.h" // PrimitiveBase
+#include <Eigen/Geometry>
+#include <Eigen/Core>
+
+namespace Ponca
+{
+
+
+
+template < class DataPoint, class _WFunctor, typename T = void  >
+class Line : public PrimitiveBase<DataPoint, _WFunctor>
+{
+private:
+
+   typedef PrimitiveBase<DataPoint, _WFunctor> Base;
+
+protected:
+
+    enum
+    {
+        PROVIDES_LINE /*!< \brief Provides  Line */
+    };
+
+public:
+
+    /*! \brief Scalar type inherited from DataPoint */
+    typedef typename DataPoint::Scalar      Scalar;
+    /*! \brief Vector type inherited from DataPoint */
+    typedef typename DataPoint::VectorType  VectorType;
+    /*! \brief Matrix type inherited from DataPoint */
+    typedef typename DataPoint::MatrixType  MatrixType;
+    /*! \brief Weight Function */
+    typedef _WFunctor                       WFunctor;
+
+
+private:
+    /*! \brief  The equation of a line with a point vectorType point 
+    and direction vectorType direction */
+
+    VectorType _direction;
+    VectorType _point; 
+
+private:
+
+    //! \brief Evaluation position (needed for centered basis)
+    VectorType m_p;
+
+public:
+
+    /*! \brief Default constructor */
+    PONCA_MULTIARCH inline Line()
+        : Base()
+    {
+        resetPrimitive();
+    }
+
+
+    /*! \brief Explicit conversion to Line, to access methods potentially hidden by inheritage */
+    PONCA_MULTIARCH inline
+    Line<DataPoint, WFunctor, T>& line()
+    { return * static_cast<Line<DataPoint, WFunctor, T>*>(this); }
+
+    /*! \brief Set the scalar field values to 0 and reset the isNormalized() status
+
+        \warning Set a_y to Zero(), which leads to nans in OrientedLine::normal()
+        \FIXME Set and use Base::m_state to handle invalid configuration
+    */
+    PONCA_MULTIARCH inline void resetPrimitive()
+    {
+        Base::resetPrimitive();
+        _direction = VectorType::Zero();
+        _point = VectorType::Zero();
+
+    }
+
+    PONCA_MULTIARCH inline bool operator==(const Line<DataPoint, WFunctor, T>& other) const{
+        return (_direction == other.direction()) && (_point == other.point());
```

Comment:
`==` operator is not reliable when comparing floating values or vectors.
Prefer https://eigen.tuxfamily.org/dox/classEigen_1_1DenseBase.html#ae8443357b808cd393be1b51974213f9c